filebeat/beater/channels.go

```diff
@@ -37,10 +40,16 @@ func newSpoolerOutlet(
 		done:    done,
 		spooler: s,
 		wg:      wg,
+		isOpen:  1,
 	}
 }
 
 func (o *spoolerOutlet) OnEvent(event *input.Event) bool {
+	open := atomic.LoadInt32(&o.isOpen) == 1
```

Comment:
It could still happen that 2 go routines both get `open == 0`, then o.done is closed and one goet into the first case, the other one into the second? Do we need a mutex?